augur/align.py

```diff
@@ -178,19 +188,34 @@ def postprocess(output_file, ref_name, keep_reference, fill_gaps):
 
 def read_sequences(*fnames):
     """return list of sequences from all fnames"""
-    seqs = {}
+    sequence_hash_by_name = {}
+
     try:
-        for fname in fnames:
-            for record in SeqIO.parse(fname, 'fasta'):
-                if record.name in seqs and record.seq != seqs[record.name].seq:
+        # Stream sequences from all input files into a single output file,
+        # skipping duplicate records (same strain and sequence) and noting
+        # mismatched sequences for the same strain name.
+        for record in io_read_sequences(*fnames):
+            # Hash each sequence and check whether another sequence with the
+            # same name already exists and if the hash is different.
+            sequence_hash = hashlib.sha256(str(record.seq).encode("utf-8")).hexdigest()
```

Comment:
In order to avoid spurious errors about mismatched duplicate sequences, should we make a best effort attempt at normalizing `record.seq` before hashing it?